tests/tasks/tkn-bundle.go

```diff
@@ -0,0 +1,395 @@
+package tasks
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/crane"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	kubeapi "github.com/redhat-appstudio/e2e-tests/pkg/clients/kubernetes"
+	"github.com/redhat-appstudio/e2e-tests/pkg/constants"
+	"github.com/redhat-appstudio/e2e-tests/pkg/framework"
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+	"github.com/tektoncd/cli/pkg/bundle"
+	pipeline "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var _ = framework.TaskSuiteDescribe("tkn bundle task", Label("tasks", "HACBS"), func() {
+
+	defer GinkgoRecover()
+
+	var namespace string
+	var kubeClient *framework.ControllerHub
+	var fwk *framework.Framework
+	var taskName string = "tkn-bundle"
+	var pvcName string = "source-pvc"
+	var pvcAccessMode corev1.PersistentVolumeAccessMode = "ReadWriteOnce"
+	var baseTaskRun *pipeline.TaskRun
+	var bundleImg string
+
+	AfterEach(framework.ReportFailure(&fwk))
+
+	BeforeAll(func() {
+		namespace = os.Getenv(constants.E2E_APPLICATIONS_NAMESPACE_ENV)
+		if len(namespace) > 0 {
+			adminClient, err := kubeapi.NewAdminKubernetesClient()
+			Expect(err).ShouldNot(HaveOccurred())
+			kubeClient, err = framework.InitControllerHub(adminClient)
+			Expect(err).ShouldNot(HaveOccurred())
+			_, err = kubeClient.CommonController.CreateTestNamespace(namespace)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// set a custom bundle repo for the task
+			bundleRepo := os.Getenv("TKN_BUNDLE_REPO")
+			Expect(bundleRepo).NotTo(BeEmpty())
+			bundleImg = fmt.Sprintf("%s:%s", bundleRepo, taskName)
+		} else {
+			var err error
+			fwk, err = framework.NewFramework(utils.GetGeneratedNamespace("konflux-task-runner"))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(fwk.UserNamespace).NotTo(BeNil(), "failed to create sandbox user")
+			namespace = fwk.UserNamespace
+			kubeClient = fwk.AsKubeAdmin
+
+			err = kubeClient.CommonController.CreateQuayRegistrySecret(namespace)
+			Expect(err).NotTo(HaveOccurred())
+			bundleImg = fmt.Sprintf("quay.io/%s/test-images:%s", utils.GetQuayIOOrganization(), taskName)
+		}
+
+		// if pvc does not exist create it
+		if _, err := kubeClient.TektonController.GetPVC(pvcName, namespace); err != nil {
+			_, err = kubeClient.TektonController.CreatePVCInAccessMode(pvcName, namespace, pvcAccessMode)
+			Expect(err).NotTo(HaveOccurred())
+			// use a pod to copy test data to the pvc
+			_, err = kubeClient.CommonController.CreatePod(setupTestData(pvcName), namespace)
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
+
+	BeforeEach(func() {
+		// if the task isn't installed on the cluster, don't run
+		if _, err := kubeClient.TektonController.GetTask(taskName, namespace); err != nil {
+			Skip("Skipping test because TKN_BUNDLE_TASK is not set")
```

Comment:
Yes, that is the plan, but I ended up creating another story for it, because I'm not sure how to handle it. The questions I have are: How do I make sure the e2e tests test the specific task and revision in the pull request? And do I have to run the tests on all pull requests even if the specific task wasn't changed?

Reply:
@joejstuart there's [this piece of code](https://github.com/redhat-appstudio/e2e-tests/blob/317aa018c6f17f0a858cbe744cf1cd636a8e2f11/tests/build/build_templates.go#L508-L532) for handling EC pipelines validation in build-definitions PR e2e test - would that perhaps help in this case?

Reply:
Yes, that's perfect. Here is the change. Thanks! https://github.com/redhat-appstudio/e2e-tests/pull/1127/commits/aded7be561b9f2a495d230293afb315c30a8d9d0